internal/system/agent/services.go

```diff
@@ -25,35 +29,349 @@ const (
 	RESTART = "restart"
 )
 
-func invokeAction(action string, services []string, params map[string]string) bool {
+type Wrappee1 ConfigRespMap
+type Wrappee2 MetricsRespMap
 
-	// Loop through requested action, along with respectively-supplied parameters.
+var responseJSON string
+var resp *http.Response
+
+func InvokeOperation(action string, services []string, params []string) bool {
+
+	// Loop through requested operation, along with respectively-supplied parameters.
 	for _, service := range services {
-		// TODO: Logging is placeholder for actual functionality to be implemented.
-		LoggingClient.Info(fmt.Sprintf("OPERATION >> About to {%v} the service {%v} with parameters {%v}.\n", action, service, params))
+		LoggingClient.Info(fmt.Sprintf("About to {%v} the service {%v} with parameters {%v}.", action, service, params))
+
+		switch action {
+
+		case STOP:
+			switch service {
+			case edgexSupportNotifications:
+				if Conf.OsLevelOperations {
+					OsExecuteStop(edgexSupportNotifications, params[0])
+				} else if Conf.DockerLevelOperations {
+					StopDockerContainer(edgexSupportNotifications)
+				}
```

Comment:
This layer of the code should be using an abstraction rather than if/else with explicit knowledge of OS vs Docker vs something else. That abstraction should be accessed via a global var that is initialized in init.go::Retry()